### PR TITLE
JAMES-3680 Document OIDC set for JMAP

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/jmap.adoc [hunk 1]
```diff
@@ -85,8 +85,21 @@ Enabling *TRACE* on `org.apache.james.jmap.wire` enables reactor-netty wiretap,
...
+
+We do supply an link:https://github.com[example] of such a setup. It combines the link:https://www.keycloak.org/[Keycloack]
+OIDC provider with the link:https://www.krakend.io/[Krackend] API gateway, but usage of similar technologies is definitely doable.
+
 == Generating a JWT key pair
 
+Apache James can alternatively be configured to check the validity of JWT tokens itself. No revocation mechanism is
+supported in such a setup, and the `sub` claim is used to identify the user. The key configuration is static.
+
+THis requires the `JWTAuthenticationStrategy` authentication strategy to be used.
```
Comment (Arsnael) [inline server/apps/distributed-app/docs/modules/ROOT/pages/configure/jmap.adoc:101]:
```suggestion
This requires the `JWTAuthenticationStrategy` authentication strategy to be used.
```

**Comment by Arsnael** [comment]
Just to clarify: what do you mean exactly on point 1. in your SSO schema by `Unauthenticated request will return HTTP authentication challenges`? Krakend doesn't redirect you or return you a challenge or anything, you just get a 401 unauthorized response from it?

**Comment by chibenwa** [comment]
>  Krakend doesn't redirect you or return you a challenge or anything, you just get a 401 unauthorized response from it?

Have you checked response headers?

**Comment by vttranlina** [comment]
> > Krakend doesn't redirect you or return you a challenge or anything, you just get a 401 unauthorized response from it?
> 
> Have you checked response headers?

I remember is 401, NOT 302, 301

**Comment by chibenwa** [comment]
https://github.com/devopsfaith/krakend-jose/issues/86

**Comment by chibenwa** [comment]
Ok, then apparently

 - 1. there seems there is not really `WWW-Authenticate` challenge for OIDC (soooo sad)
 - 2. there is an OIDC spec for discovery: https://openid.net/specs/openid-connect-discovery-1_0.html
 
 So...
 
 I will add a note that ID provider discovery is not in the scope of the example.

**Review by Arsnael — APPROVED** [review]

**Review by vttranlina — APPROVED** [review]

**Comment by Arsnael** [comment]
Can you squash the fixups? :)

**Review by quantranhong1999 — APPROVED** [review]

**Comment by chibenwa** [comment]
Needs a rebase...

**Pull request merged**